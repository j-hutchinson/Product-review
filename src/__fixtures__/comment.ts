import { Comment } from '../types';

export const singleComment: Comment = {
	name: 'Jack',
	email: 'something@aol.com',
	rating: 2,
	comment: 'adequate product'
};
