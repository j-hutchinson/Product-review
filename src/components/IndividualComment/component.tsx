import React from 'react';
import styled from 'styled-components';
import { Comment } from '../../types';

interface Props {
    comment: Comment
}

const StyledComment = styled.div`
    padding: 20px;
`;

const StyledLabel = styled.div``;

const IndividualComment = ({ comment }: Props) => {
    return (
        <StyledComment>
            <StyledLabel>Name: {comment.name}</StyledLabel>
            <StyledLabel>Rating: {comment.rating}</StyledLabel>
            <StyledLabel>Email: {comment.email}</StyledLabel>
            <StyledLabel>Comment: {comment.comment}</StyledLabel>
        </StyledComment>
    )
}

export default IndividualComment;