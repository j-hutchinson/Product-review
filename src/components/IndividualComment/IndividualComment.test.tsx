import { shallow } from 'enzyme';
import React from 'react';
import IndividualComment from './component';
import { singleComment } from '../../__fixtures__/comment'

describe('IndividualComment component', () => {
    test('component matches snapshot', () => {
        expect.assertions(1);

        const wrapper = shallow(<IndividualComment comment={singleComment} />);

        expect(wrapper).toMatchSnapshot();
    });
});