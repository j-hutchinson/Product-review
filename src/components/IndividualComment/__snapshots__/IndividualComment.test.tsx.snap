// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndividualComment component component matches snapshot 1`] = `
<styled.div>
  <styled.div>
    Name: 
    Jack
  </styled.div>
  <styled.div>
    Rating: 
    2
  </styled.div>
  <styled.div>
    Email: 
    something@aol.com
  </styled.div>
  <styled.div>
    Comment: 
    adequate product
  </styled.div>
</styled.div>
`;
